packages/core/scripts/local/ClaimAllRewards.js

```diff
@@ -1,72 +1,190 @@
     const voter = event.returnValues.voter;
-    const newPriceRequest = { identifier: event.returnValues.identifier, time: event.returnValues.time };
-    if (votersToPriceRequests[voter]) {
-      votersToPriceRequests[voter].push(newPriceRequest);
+    const round = event.returnValues.roundId;
+    const contractAddress = event.address;
+    const key = [voter, round, contractAddress].join("|");
+    if (excludeAddresses.includes(toChecksumAddress(voter))) {
+      console.log("Found noclaim voter", voter);
+      continue;
+    }
+    const newPriceRequest = {
+      identifier: event.returnValues.identifier,
+      time: event.returnValues.time,
+      ancillaryData: event.returnValues.ancillaryData === null ? "0x" : event.returnValues.ancillaryData
+    };
+    if (priceRequestMap[key]) {
+      priceRequestMap[key].push(newPriceRequest);
     } else {
-      votersToPriceRequests[voter] = [newPriceRequest];
+      priceRequestMap[key] = [newPriceRequest];
     }
   }
 
   const retrievableRewards = (
     await Promise.all(
-      Object.entries(votersToPriceRequests).map(async ([voter, priceRequests]) => {
+      Object.entries(priceRequestMap).map(async ([key, priceRequests]) => {
+        const [voter, round, contractAddress] = key.split("|");
+        const voting = votingContracts.find(
+          contract => toChecksumAddress(contract.options.address) === toChecksumAddress(contractAddress)
+        );
+        if (!voting) throw `Couldn't find voting contract for ${contractAddress}`;
+        const fullVotingContract = new web3.eth.Contract(getAbi("Voting", "latest"), voting.options.address);
+        const migratedAddress = await fullVotingContract.methods.migratedAddress().call();
         try {
-          const output = await voting.retrieveRewards.call(voter, argv.round, priceRequests);
+          const output = await voting.methods
+            .retrieveRewards(voter, round, priceRequests)
+            .call({ from: migratedAddress });
           if (output.toString() === "0") {
             return null;
           } else if (toBN(output.toString()).gt(toBN(toWei("100000000")))) {
             // If the output is bigger than 100MM tokens, that means this is _really_ a revert.
             return null;
           } else {
             console.log("Found Rewards for voter", voter);
-            return [voter, priceRequests];
+            return { voter, priceRequests, round, voting, rewards: toBN(output.toString()) };
           }
         } catch (error) {
+          console.error(error);
           return null;
         }
       })
     )
   ).filter(element => element !== null);
 
-  const dataArray = retrievableRewards.map(([voter, priceRequests]) => {
-    return voting.contract.methods.retrieveRewards(voter, argv.round, priceRequests).encodeABI();
+  if (argv.onlyCompute) {
+    const groupedByVoter = lodash.groupBy(retrievableRewards, el => el.voter);
+    Object.entries(groupedByVoter).forEach(([voter, rewardArray]) => {
```

Comment:
o boy. what a console log.

Reply:
this is interesting, never heard of console.group!